apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -303,26 +302,26 @@ defmodule Aecore.Miner.Worker do
     )
   end
 
-  @spec generate_and_add_micro_block(
-          Chainstate.t(),
-          non_neg_integer(),
-          binary(),
-          binary(),
-          non_neg_integer(),
-          boolean()
-        ) :: :ok | {:error, String.t()}
-  def generate_and_add_micro_block(
-        chain_state,
-        top_height,
-        prev_hash,
-        prev_key_hash,
-        last_time,
-        loop_micro_blocks
-      ) do
+  @spec generate_and_add_micro_block(boolean()) :: :ok | {:error, String.t()}
+  def generate_and_add_micro_block(loop_micro_blocks) do
+    top_block = Chain.top_block()
+    prev_hash = Header.hash(top_block.header)
+    prev_key_hash = Header.top_key_block_hash(top_block.header)
+
+    minimum_distance =
+      if prev_hash == prev_key_hash do
+        @minimum_distance_from_key_block
+      else
+        GovernanceConstants.micro_block_distance()
+      end
+
+    :timer.sleep(minimum_distance)
```

Comment:
This function is called by `Channel.Worker` I think we shouldn't have any "sleeps" here, because this would block the Channel.Worker. It would be better to have such stuff in some thread maybe Miner GenServer. (make a cast on the genserver, cast-s don't block) Generally handling of microblock mining should be improved. Probably not in this PR. Probably it would be better to have a separate module from that, as mining MicroBlocks is something completely different then mining KeyBlocks.

Reply:
I've thought about having a separate module for micro block generation, decided that it could be done a bit later on as it works this way as well. And as for the sleep, maybe it should be done before sending the new block to our peers.